pkg/clusterregistry/options/options.go

```diff
@@ -26,31 +26,38 @@ import (
 	"github.com/spf13/pflag"
 )
 
-// Runtime options for the clusterregistry-apiserver.
+// Runtime options for the cluster registry.
 type ServerRunOptions struct {
-	GenericServerRunOptions *genericoptions.ServerRunOptions
-	Etcd                    *genericoptions.EtcdOptions
-	SecureServing           *genericoptions.SecureServingOptions
-	Audit                   *genericoptions.AuditOptions
-	Features                *genericoptions.FeatureOptions
-	Authentication          *BuiltInAuthenticationOptions
+	GenericServerRunOptions  *genericoptions.ServerRunOptions
+	Etcd                     *genericoptions.EtcdOptions
+	SecureServing            *genericoptions.SecureServingOptions
+	Audit                    *genericoptions.AuditOptions
+	Features                 *genericoptions.FeatureOptions
+	StandaloneAuthentication *StandaloneAuthenticationOptions
+	StandaloneAuthorization  *StandaloneAuthorizationOptions
+	DelegatingAuthentication *genericoptions.DelegatingAuthenticationOptions
+	DelegatingAuthorization  *genericoptions.DelegatingAuthorizationOptions
 
-	EventTTL time.Duration
+	EventTTL         time.Duration
+	UseDelegatedAuth bool
 }
 
 // NewServerRunOptions creates a new ServerRunOptions object with default values.
 func NewServerRunOptions() *ServerRunOptions {
 	o := &ServerRunOptions{
 		GenericServerRunOptions: genericoptions.NewServerRunOptions(),
-		Etcd:           genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
-		SecureServing:  genericoptions.NewSecureServingOptions(),
-		Audit:          genericoptions.NewAuditOptions(),
-		Features:       genericoptions.NewFeatureOptions(),
-		Authentication: NewBuiltInAuthenticationOptions().WithAll(),
+		Etcd:                     genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
+		SecureServing:            genericoptions.NewSecureServingOptions(),
+		Audit:                    genericoptions.NewAuditOptions(),
+		Features:                 genericoptions.NewFeatureOptions(),
+		StandaloneAuthentication: NewStandaloneAuthenticationOptions().WithAll(),
+		StandaloneAuthorization:  NewStandaloneAuthorizationOptions(),
+		DelegatingAuthentication: genericoptions.NewDelegatingAuthenticationOptions(),
+		DelegatingAuthorization:  genericoptions.NewDelegatingAuthorizationOptions(),
 
-		EventTTL: 1 * time.Hour,
+		EventTTL:         1 * time.Hour,
+		UseDelegatedAuth: false,
```

Comment:
I feel like most may want to deploy the CR using the k8s aggregator and enable delegated auth by default. We can always change this in the future and if using `crinit` then it doesn't really matter too much.

Reply:
Good point. Done.

Reply:
?

Reply:
Hmm. I meant to change this to `true`, let me fix that now.